src/Bicep.Core/Semantics/ParamAssignmentSymbolVisitor.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Extensions;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem;
+using Bicep.Core.Workspaces;
+
+namespace Bicep.Core.Semantics
+{
+    public sealed class ParamAssignmentSymbolVisitor : SyntaxVisitor
```

Comment:
Just realized the name might be a bit confusing since there is `SymbolVisitor`.

```suggestion
    public sealed class ParamAssignmentSymbolCollectVisitor : SyntaxVisitor
```